lib/utils.js

```diff
@@ -55,22 +55,23 @@ function mergeObject(target, source, shouldRemoveNullObjectValues = true) {
     for (let i = 0; i < sourceKeys.length; ++i) {
         const key = sourceKeys[i];
 
-        // If we pass undefined as the updated value for a key, we want to generally ignore it
         // If shouldRemoveNullObjectValues is true, we want to remove null values from the merged object
-        if (source[key] === undefined || (shouldRemoveNullObjectValues && source[key] === null)) {
-            // eslint-disable-next-line no-continue
-            continue;
-        }
+        const shouldOmitSourceKey = shouldRemoveNullObjectValues && source[key] === null;
 
-        if (!isMergeableObject(source[key]) || !target[key]) {
-            if (targetAndSourceIdentical) {
-                // eslint-disable-next-line no-continue
-                continue;
+        // If we pass undefined as the updated value for a key, we want to generally ignore it
+        const isSourceKeyUndefined = source[key] === undefined;
+
+        if (!isSourceKeyUndefined && !shouldOmitSourceKey) {
+            const isSourceKeyMergable = isMergeableObject(source[key]);
+
+            if (isSourceKeyMergable && target[key]) {
+                if ((!shouldRemoveNullObjectValues || isSourceKeyMergable)) {
+                    // eslint-disable-next-line no-use-before-define
```

Comment:
NAB: I see this was really just moved, but can this `disable` also be removed if `fastMerge` is moved to the top of the file?

Reply:
not really, because `fastMerge` uses `mergeObject` and vice-versa...

Reply:
Ah, OK. Makes sense then that one of them would need this. yay for JS hoisting!